### PR TITLE
feat: add slitherin 

### LINKED ISSUE #196: [Feature Request]: Add slitherin check to CI workflows
### Summary

https://github.com/pessimistic-io/slitherin

### Problem

Extra set of the Slither detectors from the [pessimistic.io](https://pessimistic.io/) team.

### Implementation

_No response_

### Guidelines

- [X] I agree to follow [Code of Conduct](https://github.com/lidofinance/core/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read [Lido Contribution Guide](https://github.com/lidofinance/core/blob/master/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by tamtamchik**
Fixes #196 

Excluded detectors:
- `pess-strange-setter` produces an error: 
  - `AttributeError: 'NewElementaryType' object has no attribute '_expression'`
-  `pess-arbitrary-call-calldata-tainted` produces an error on SARIF file generation (https://github.com/crytic/slither/pull/2432)

### CODE REVIEW

**Review by rnmsslido — APPROVED** [review]
LGTM

**Comment by tamtamchik** [comment]
@rnmsslido WOW, that was fast! Thank you, sir!

File: .github/workflows/analyse.yml [hunk 1]
```diff
@@ -20,14 +20,12 @@ jobs:
       # REVIEW: here and below steps taken from official guide
       # https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages
```
Comment (TheDZhon) [inline .github/workflows/analyse.yml:21]: the link is about the caching techniques but I see no explicit mentions of cache :thinking:

File: .github/workflows/analyse.yml
Comment (TheDZhon) [inline .github/workflows/analyse.yml:1]: Are these review notes still sound?

**Review by TheDZhon — COMMENTED** [review]
LGTM mostly, some concerns only about the comments we have

**Pull request merged**